python_scripts/trees.py

```diff
@@ -362,24 +363,26 @@ def classification_criterion(labels):
 
 # %% [markdown]
 # In our case, we can see that the entropy in the partition < 42 mm is close to
-# 0 meaning that this partition is "pure" and contain a single class while
-# the partition >= 42 mm is much higher due to the fact that 2 of the classes
-# are still mixed.
+# 0, meaning that this partition is "pure" and nearly entirely consists of a
+# single class (Adelie). Conversely, the partition >= 42 mm is much higher
+# because the species are still mixed, with large numbers of both Chinstrap
+# and Gentoo penguins.
 #
-# Now, we are able to assess the quality of each partition. However, the
-# ultimate goal is to evaluate the quality of the split and thus combine both
-# measures of entropy to obtain a single statistic.
+# With entropy, we are able to assess the quality of each partition. However,
+# the ultimate goal is to evaluate the quality of the over all split and thus
+# combine the measures of entropy in each partition into a single statistic.
 #
 # ### Information gain
 #
-# This statistic is known as the information gain. It combines the entropy of
-# the different partitions to give us a single statistic qualifying the quality
-# of a split. The information gain is defined as the difference of the entropy
-# before making a split and the sum of the entropies of each partition,
-# normalized by the frequencies of class samples on each partition. The goal is
+# Information gain uses the entropy of
+# the two partitions to give us a single statistic quantifying the quality
+# of a split. The information gain is defined as the difference between the
+# entropy
+# before a split and the sum of the entropies of each partition,
+# normalized by the frequencies of class samples in each partition. The goal is
```

Comment:
It might be nice to give a formula for this? I see it's clear from the code what is happening but I think just a formula (and even an example calculation) means people don't have to read the code.